### ISSUE TITLE
Status of picture/icon methods in lib/gpx.rb

### ISSUE THREAD

**Issue by gravitystorm** [openstreetmap/openstreetmap-website#281]
The two methods mentioned try calling "Magick::Image" and "Magick::ImageList", which I guess are supposed to be RMagick calls. And I also guess that since rmagick was removed some time ago, these calls aren't actually being used in production, i.e. the standalone gpx-importer is making the images.

So as written, these methods throw errors when called from the console.

```
gpx = GPX::File.new('')
gpx.picture(0,0,1,1,10)
NameError: uninitialized constant GPX::File::Magick
```

Should we keep these methods, write some tests for them and fix the rmagick dependency? This way, developers could upload GPX traces for testing, without needing to install the gpx-importer. Or should we remove them, since the code isn't used in production, they don't work, and they don't have any tests?

**Comment by mmd-osm** [comment]
Just a thought: could we backport the image.c code in gpx-import to ruby, using https://github.com/dark-panda/gd2-ffij (or maybe the original gd2 library)? This would only introduce a dependency on gd2, rather than some Imagemagick stuff.

https://github.com/openstreetmap/gpx-import/blob/master/src/image.c

**Comment by mmd-osm** [comment]
Rendering a GPX trace icon isn't too difficult using gd2-ffij:

```ruby
    require 'gd2-ffij'

[...]

    def icon(min_lat, min_lon, max_lat, max_lon)
      width = 50
      height = 50
      proj = OSM::Mercator.new(min_lat, min_lon, max_lat, max_lon, width, height)

      image = GD2::Image::IndexedColor.new(width, height)

      black = image.palette.allocate(GD2::Color[0, 0, 0])
      white = image.palette.allocate(GD2::Color[255, 255, 255])

      image.draw do |pen|
        pen.color = white
        pen.rectangle(0, 0, width, height, true)
      end

      image.draw do |pen|
        pen.color = black
        pen.anti_aliasing = true
        pen.dont_blend = false

        oldpx = 0.0
        oldpy = 0.0

        first = true

        points do |p|
          px = proj.x(p.longitude)
          py = proj.y(p.latitude)

          pen.line(px, py, oldpx, oldpy) unless first

          first = false
          oldpy = py
          oldpx = px
        end

      end
  
      image.gif
    end
  end
```

**Comment by gravitystorm** [comment]
This has been resolved by #2120, which now uses the [gd2-ffij](https://github.com/dark-panda/gd2-ffij) library for creating pictures and icons.
